daprdocs/content/en/operations/components/pluggable-components/pluggable-components-discovery.md

```diff
@@ -0,0 +1,212 @@
+
+## Step 2: Declaring a gRPC-based Pluggable Component
+
+gRPC-based Pluggable Components are defined using the [Component Spec]({{< ref component-schema.md >}}) and its `type` is derived from the socket name (without the file extension).
+
+Place the following file in the defined components-path (replace `your_socket_goes_here` by your component socket name without any extension and `your_component_type` by your component type).
+
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: prod-mystore
+spec:
+  type: your_component_type.your_socket_goes_here
+  version: v1
+  metadata:
+```
+
+Using `memstore.sock` as a concrete example we have the following, and assuming that `memstore` implements the StateStoreService.
+
+- `your_component_type` would be replaced by `state`, as it is a state store.
+- `your_socket_goes_here` would be replaced by `memstore`: `memstore.sock` without any extension.
+
+The full configuration for `memstore` would be:
+
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: prod-mystore
+spec:
+  type: state.memstore
+  version: v1
+  metadata:
+```
+
+Save this file as `component.yaml` in Dapr's configuration folder.
+
+## Step 3: Running Dapr
+
+Initialize Dapr by following the [tutorial]({{< ref get-started-api.md >}}), and make sure that your component Spec is placed in the right folder.
+
+> Note: 1.9.0 is the minimum Dapr version that supports gRPC-based Pluggable Components
+> specify the runtime version by using `--runtime-version` flag
+
+<!-- We should list the actual command line the user will be typing here -->
+
+That's it! Now we are able to call the statestore APIs via Dapr API.
+
+See it working by running
+
+```shell
+curl -X POST -H "Content-Type: application/json" -d '[{ "key": "name", "value": "Bruce Wayne", "metadata": {}}]' http://localhost:$PORT/v1.0/state/prod-mystore
+```
+
+Retrieve the value
+
+```shell
+curl http://localhost:$PORT/v1.0/state/prod-mystore/name
+```
+
+> replace $PORT by Dapr http port
+
+{{% /codetab %}}
+
+{{% codetab %}}
+
+## Step 1: Build and Publish a container for your gRPC-based Component
+
+As a prerequisite for running on kubernetes mode, your component must run as a container. Which means that it should be published first and accessible by your kubernetes cluster.
+
+## Step 2: Deploying Dapr on a Kubernetes cluster
+
+Follow the steps provided in the [Deploy Dapr on a Kubernetes cluster]({{< ref kubernetes-deploy.md >}}) docs and make sure you have a kubernetes cluster configured with Dapr version 1.9+.
+
+## Step 3: Adding Pluggable Component Container in your Deployments
+
+When running on kubernetes mode, gRPC-based Pluggable Components are side-car containers.
+
+As gRPC-based Pluggable Components are backed by [Unix Domain Sockets][uds], the first step is configuring the deployment spec to ensure that the socket created by your Pluggable Component are accessible by Dapr runtime, to do that, you have to:
```

Comment:
```suggestion
Since gRPC-based pluggable components are backed by [Unix Domain Sockets][uds], make the socket created by your pluggable component accessible by Dapr runtime. Configure the deployment spec:
```